### PR TITLE
Initial support for FlexHTx

### PR DESCRIPTION
**Pull request by danielhstahl**
This is a small PR that adds HDMI and the Flex HT HWID and DSP version.  This allows for minidsp-rs to recognize the Flex HT and switch inputs correctly.  Similar approach to https://github.com/mrene/minidsp-rs/issues/566#issuecomment-2140892908.  

Long term, I'd like to add broader support but the XML from the Flex HT seems quite distinct from the other MiniDSP offerings and I'm not familiar enough with the "Input" "Output" architecture to update it in a way that still accommodates the other MiniDSP products.  See https://github.com/danielhstahl/minidsp-rs/blob/feature/add_flex_ht/devtools/src/codegen/flexht/mod.rs for a first attempt.

I'd also like a way to retrieve the HDMI information which currently shows up in the Device Console:
image

but verbose logging while connecting via the minidsp service does not show anything obvious beyond the input/output levels:

[131.972s] Recv: FloatData(FloatView { base: 330, data: [-131.29065, -129.73033, -120.0, -133.86212, -125.83957, -126.13447, -134.42488, -130.44986] })
[132.174s] Sent: [14, 03, 10, 08]
[132.178s] Sent: ReadFloats { addr: 310, len: 08 } (0x310 | 784) <> <unknown>
[132.178s] Recv: [14, 03, 10, d5, 59, 03, c3, d5, 59, 03, c3, d5, 59, 03, c3, d5, 59, 03, c3, d5, 59, 03, c3, d5, 59, 03, c3, d5, 59, 03, c3, d5, 59, 03, c3]
[132.180s] Recv: FloatData(FloatView { base: 310, data: [-131.3509, -131.3509, -131.3509, -131.3509, -131.3509, -131.3509, -131.3509, -131.3509] })
[132.181s] Sent: [14, 03, 30, 08]
[132.182s] Sent: ReadFloats { addr: 330, len: 08 } (0x330 | 816) <> <unknown>

### CODE REVIEW

**Comment by danielhstahl** [comment]
@jprouty since he also seems to own the flexht

File: protocol/Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,19 @@ thiserror = "1.0.56"
...
+    "device_10x10hd",
+    "device_4x10hd",
+    "device_shd",
+    "device_ddrc24",
+    "device_nanodigi2x8",
+    "device_ddrc88bm",
+    "device_c8x12v2",
+    "device_m2x4",
+    "device_flex",
+]
```
Comment (jprouty) [inline protocol/Cargo.toml:35]: Should the newly added `device_flex_ht` be listed here?

File: protocol/src/device/probe.rs [hunk 2]
```diff
@@ -74,6 +76,10 @@ pub fn probe_kind(device_info: &DeviceInfo) -> DeviceKind {
         (2, 22) => M2x4,
         #[cfg(feature = "device_flex")]
         (27, 100) => Flex,
+
+        #[cfg(feature = "device_flex_ht")]
+        (32, 113) => Flex, //get from `minidsp probe`
```
Comment (jprouty) [inline protocol/src/device/probe.rs]: The HTx appears to share the same device id. I cannot think of a brief way to make that evident in the device name itself - however, would you mind adding a comment here (or somewhere) that  `device_flex_ht` also supports Flex HTx?

File: protocol/Cargo.toml (hunk 1 above)
Comment (danielhstahl) [reply protocol/Cargo.toml:35]: Yes, added

File: protocol/src/device/probe.rs (hunk 2 above)
Comment (danielhstahl) [reply protocol/src/device/probe.rs]: Great point, I actually have the HTx not the HT.  I've updated the name to reflect

**Comment by danielhstahl** [comment]
I think I'm making some progress on the codegen component as well: https://github.com/danielhstahl/minidsp-rs/blob/feature/add_flex_ht/protocol/src/device/flexhtx.rs.  I'll PR that separately: https://github.com/mrene/minidsp-rs/pull/665

**Comment by Jacob-Bishop** [comment]
@danielhstahl Where did you end up getting on this? Did you run into blockers, or just run out of time/interest/energy?

**Comment by danielhstahl** [comment]
> @danielhstahl Where did you end up getting on this? Did you run into blockers, or just run out of time/interest/energy?

I couldn't get it to compile for ARM.  The original action is broken.  I ended up using the latest official release and just making calls to it: https://github.com/danielhstahl/minidsp-remote-control/blob/main/server/index.js#L65

Once Github starts supporting native ARM runners (currently in beta: https://github.blog/changelog/2024-06-03-actions-arm-based-linux-and-windows-runners-are-now-in-public-beta/) I'll revisit.

**Comment by 3ll3d00d** [comment]
@danielhstahl  I made some tweaks to the build which worked for arm last time I tried it, you can see the changes in https://github.com/3ll3d00d/minidsp-rs/commit/e09b3f22dfde601014e391d60e3dde238f62b262 in case it helps

**Comment by danielhstahl** [comment]
@3ll3d00d it doesn't seem to work for me now: https://github.com/danielhstahl/minidsp-2/actions/runs/11535095677/job/32109809012, but I appreciate pointing me to your code.

**Pull request closed without merging**